src/main/java/de/tum/in/www1/artemis/web/rest/ExamSessionResource.java

```diff
@@ -0,0 +1,99 @@
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import de.tum.in.www1.artemis.domain.exam.ExamSession;
+import de.tum.in.www1.artemis.domain.exam.StudentExam;
+import de.tum.in.www1.artemis.service.ExamSessionService;
+import de.tum.in.www1.artemis.service.StudentExamService;
+import de.tum.in.www1.artemis.service.UserService;
+
+/**
+ * REST controller for managing ExamSessions.
+ */
+@RestController
+@RequestMapping("/api")
+public class ExamSessionResource {
+
+    private final Logger log = LoggerFactory.getLogger(ExamSessionResource.class);
+
+    private final ExamSessionService examSessionService;
+
+    private final UserService userService;
+
+    private final StudentExamService studentExamService;
+
+    public ExamSessionResource(ExamSessionService examSessionService, UserService userService, StudentExamService studentExamService) {
+        this.examSessionService = examSessionService;
+        this.userService = userService;
+        this.studentExamService = studentExamService;
+
+    }
+
+    /**
+     * GET /courses/{courseId}/exams/{examId}/currentSession : Get current exam session for given exam and user
+     *
+     * @param courseId  the course to which the student exams belong to
+     * @param examId    the exam to which the student exams belong to
+     * @return the ResponseEntity with status 200 (OK) and the token as string
+     */
+    @GetMapping("/courses/{courseId}/exams/{examId}/currentSession")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<ExamSession> getCurrentExamSession(@PathVariable Long courseId, @PathVariable Long examId) {
+        var userId = userService.getUser().getId();
+        log.debug("REST request to get the current exam session for user : {} in the given exam : {}", userId, examId);
+
+        StudentExam studentExam = studentExamService.findOneByExamIdAndUserId(examId, userId);
+
+        ExamSession examSession = examSessionService.getCurrentExamSession(studentExam);
+        return ResponseEntity.ok(examSession);
+    }
+
+    /**
+     * GET /courses/{courseId}/exams/{examId}/newSession : Create exam session for given exam and user
+     *
+     * @param courseId  the course to which the student exams belong to
+     * @param examId    the exam to which the student exams belong to
+     * @return the ResponseEntity with status 200 (OK) and the token as string
+     */
+    @GetMapping("/courses/{courseId}/exams/{examId}/newSession")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<ExamSession> createExamSession(@PathVariable Long courseId, @PathVariable Long examId) {
+        var userId = userService.getUser().getId();
```

Comment:
this REST call should not be necessary, because the new session should only be created when the user starts the exam